typesafe-magnolia-tests/src/test/scala/zio/config/typesafe/TypesafeConfigOptionalTest.scala

```diff
@@ -782,14 +759,8 @@ object TypesafeConfigOptionalTest
             equalTo(
               Left(
                 List(
-                  MissingValue(List(Key("a"), Key("c"), Key("b2")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("c"), Key("a2")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("c"), Key("b1")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("c"), Key("a1")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("a"), Key("b2")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("a"), Key("a2")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("a"), Key("b1")), List("optional value", "value of type string")),
-                  MissingValue(List(Key("a"), Key("a"), Key("a1")), List("optional value", "value of type string"))
+                  MissingValue(List(Key("a"), Key("c")), List("optional value")),
+                  MissingValue(List(Key("a"), Key("a")), List("optional value"))
```

Comment:
This is good in the sense that the `optional` value is tagged at the root level instead of leaves as in the current version.

I am not entirely sure if its a good idea to avoid telling the user the entire tree that was missing instead of giving a hint that `a.c` and `a.a` is missing.

```scala

Right(ReadError:
╥
╠══╦══╗
║  ║  ║
║  ║  ╠══╦══╗
║  ║  ║  ║  ║
║  ║  ║  ║  ╠─MissingValue
║  ║  ║  ║  ║ path: a.c.b1
║  ║  ║  ║  ║ Details: optional value, value of type string
║  ║  ║  ║  ▼
║  ║  ║  ║
║  ║  ║  ╠─MissingValue
║  ║  ║  ║ path: a.c.a1
║  ║  ║  ║ Details: optional value, value of type string
║  ║  ║  ▼
║  ║  ║
║  ║  ╠══╦══╗
║  ║  ║  ║  ║
║  ║  ║  ║  ╠─MissingValue
║  ║  ║  ║  ║ path: a.c.b2
║  ║  ║  ║  ║ Details: optional value, value of type string
║  ║  ║  ║  ▼
║  ║  ║  ║
║  ║  ║  ╠─MissingValue
║  ║  ║  ║ path: a.c.a2
║  ║  ║  ║ Details: optional value, value of type string
║  ║  ║  ▼
║  ║  ▼
║  ║
║  ╠══╦══╗
║  ║  ║  ║
║  ║  ║  ╠─MissingValue
║  ║  ║  ║ path: a.a.b1
║  ║  ║  ║ Details: optional value, value of type string
║  ║  ║  ▼
║  ║  ║
║  ║  ╠─MissingValue
║  ║  ║ path: a.a.a1
║  ║  ║ Details: optional value, value of type string
║  ║  ▼
║  ║
║  ╠══╦══╗
║  ║  ║  ║
║  ║  ║  ╠─MissingValue
║  ║  ║  ║ path: a.a.b2
║  ║  ║  ║ Details: optional value, value of type string
║  ║  ║  ▼
║  ║  ║
║  ║  ╠─MissingValue
║  ║  ║ path: a.a.a2
║  ║  ║ Details: optional value, value of type string
║  ║  ▼
║  ▼
▼)


```